### PR TITLE
Return the current agent state on Fixture.IsHealthy method

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

Modifies `Fixture.IsHealthy` to return the current agent status if the agent isn't healthy.

## Why is it important?

When debugging a failing test it's valuable to know the current status of the agent.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## Related issues

- N/A

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: pkg/testing/fixture.go [hunk 1]
```diff
@@ -724,16 +724,23 @@ func (f *Fixture) ExecVersion(ctx context.Context, opts ...process.CmdOption) (A
...
 func (f *Fixture) IsHealthy(ctx context.Context, opts ...process.CmdOption) (bool, error) {
 	status, err := f.ExecStatus(ctx, opts...)
 	if err != nil {
 		return false, fmt.Errorf("agent status returned and error: %w", err)
 	}
 
-	return status.State == int(cproto.State_HEALTHY), nil
+	if status.State != int(cproto.State_HEALTHY) {
+		return false, fmt.Errorf("agent isn't health, current status: %s",
+			client.State(status.State))
```
Comment (blakerouse) [inline pkg/testing/fixture.go:740]: I understand the reason for adding this, but this kinda changes the meaning of the error in this function. The error was more that it failed to get the actual health. Now this changes it to return an error when its not healthy, that is probably not was the caller is expecting.

Comment (AndersonQ) [reply pkg/testing/fixture.go:740]: This is true, but it's also why I changed it docs. Besides there is just one usage of it, that is the test which needed this change. I could change the function signature, but I believe the doc change clarifies what is returned in the error.

Comment (blakerouse) [reply pkg/testing/fixture.go:740]: Based on the way your changing it, would you be willing to change it to `EnsureHealthy` and remove the boolean? Doesn't seem like true/false is relevant to what this function is even doing at more, so just have it return error or nil.

File: pkg/testing/fixture.go [hunk 2]
```diff
@@ -724,16 +724,23 @@ func (f *Fixture) ExecVersion(ctx context.Context, opts ...process.CmdOption) (A
...
+// This function is compatible with any Elastic Agent version 8.6 or later.
 func (f *Fixture) IsHealthy(ctx context.Context, opts ...process.CmdOption) (bool, error) {
 	status, err := f.ExecStatus(ctx, opts...)
 	if err != nil {
 		return false, fmt.Errorf("agent status returned and error: %w", err)
 	}
 
-	return status.State == int(cproto.State_HEALTHY), nil
+	if status.State != int(cproto.State_HEALTHY) {
+		return false, fmt.Errorf("agent isn't health, current status: %s",
```
Comment (ycombinator) [inline pkg/testing/fixture.go:739]: Nit: typo.
```suggestion
		return false, fmt.Errorf("agent isn't healthy, current status: %s",
```

File: pkg/testing/fixture.go (hunk 1 above)
Comment (AndersonQ) [reply pkg/testing/fixture.go:740]: It does make sense, I just kept the name because it isn't ensuring anything, just checking the status

**Review by blakerouse — APPROVED** [review]
+1

**Pull request merged**